app/models/blueprint.rb

```diff
@@ -200,4 +215,37 @@ def duplicate_custom_button_sets(old_template, new_template)
       old_button_set.deep_copy(:owner => new_template)
     end
   end
+
+  def publish_bundle
+    # A mapping of what have been copied when publish the blueprint
+    {
+      "service_dialog"    => publish_resource_actions,
+      "service_templates" => publish_service_resources
+    }
+    # TODO: publish custom_buttons and custom_button_sets
+  end
+
+  def publish_resource_actions
+    dialog_map = {}
+    bundle.resource_actions.each do |action|
+      action.dialog = ensure_new_dialog(dialog_map, action.dialog)
+    end
+    dialog_map.update(dialog_map) { |_key, val| val.id }
+  end
+
+  def publish_service_resources
+    changes = {}
+    bundle.service_resources.each do |sr|
```

Comment:
Use `each_with_object({}) do |sr, changes|` should allow you to remove the `changes` variable at the top and bottom of this method.